File: gunhoflash/group-anagrams.py
from typing import List

class Solution:
  def getHash(self, s: str):
    chars = list(s)
    chars.sort()
    return ''.join(chars)

  def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
    hashMap = {}

    for s in strs:
      h = self.getHash(s)
      if not h in hashMap:
        hashMap.setdefault(h, [])
      hashMap.get(h).append(s)

    return hashMap.values()

File: gunhoflash/group-anagrams.ts
function hash(str: string): string {
  return str.split('').sort().join('');
}

function groupAnagrams(strs: string[]): string[][] {
  const map = new Map<string, string[]>();

  for (const str of strs) {
    const h = hash(str);
    if (!map.has(h)) {
      map.set(h, []);
    }
    map.get(h)!.push(str);
  }

  return Array.from(map.values());
}
